documentation/specs/CodeVisualizer.md

```diff
@@ -0,0 +1,135 @@
+# `CodeVisualizer` Component Specification
+
+## Design
```

Comment:
the design for this was heavily inspired by Twilio's: https://paste.twilio.design/components/code-block#tabs